### ISSUE TITLE
Using C++ from Cocoapod

### ISSUE THREAD

**Issue by ospfranco** [matrix-org/olm#89]
I have a iOS project with the pod depedendency of Olm:

```ruby
s.dependency "OLMKit", "3.2.12"
```

However, I do not want to use the objective-C bindings, but the C++ ones. However, on my C++ code when I try to import:

```
#include <OLMKit/olm.h>
```

I get a header error:

![Oscar Franco Screen 000984](https://github.com/matrix-org/olm/assets/1634213/566abbb8-f875-4646-a128-7fa7fc42cab1)

Is there any way to achieve this?

**Comment by ospfranco** [comment]
It also seems the latest version on cocoapods is [3.2.12](https://github.com/CocoaPods/Specs/blob/master/Specs/2/8/0/OLMKit/3.2.12/OLMKit.podspec.json) but the latest available version is [3.2.16](https://gitlab.matrix.org/matrix-org/olm/-/blob/master/OLMKit.podspec?ref_type=heads)? Could it be posible to publish the latest version?

**Comment by ospfranco** [comment]
I think the problem is this [podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/2/8/0/OLMKit/3.2.12/OLMKit.podspec.json) has hardcoded paths to `/Users/stefanceriu/Developer/olm/include` right? so Xcode doesn't know where to find the actual include folder. @stefanceriu could you publish an fix/update?

**Comment by ospfranco** [comment]
[After reading up a little bit](https://stackoverflow.com/questions/70360557/podspec-file-keep-folder-structure-of-public-c-headers), it seems xcode/clang doesn't support folder structures, which would explain why this includes will not work.

I modified the user header search paths manually and making them `recursive` in the hopes it would work:

![Oscar Franco Screen 000985](https://github.com/matrix-org/olm/assets/1634213/73fe2c24-abf4-4721-8bde-8671616a09b1)

But alas the error remains, so it seems this won't be possible (despite the errors in the podspec).

Would be nice to get some confirmation this assumption is correct.

**Comment by uhoreg** [comment]
> It also seems the latest version on cocoapods is [3.2.12](https://github.com/CocoaPods/Specs/blob/master/Specs/2/8/0/OLMKit/3.2.12/OLMKit.podspec.json) but the latest available version is [3.2.16](https://gitlab.matrix.org/matrix-org/olm/-/blob/master/OLMKit.podspec?ref_type=heads)? Could it be posible to publish the latest version?

The libolm releases since 3.2.12 don't have any changes that affect the iOS bindings.

**Comment by stefanceriu** [comment]
> [podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/2/8/0/OLMKit/3.2.12/OLMKit.podspec.json) has hardcoded paths to /Users/stefanceriu/Developer/olm/include

That's super weird, I think CocoaPods just decided to do that on its own as I `pod trunk push`-ed it from my machine. 
The [original podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/2/8/0/OLMKit/3.2.12/OLMKit.podspec.json) is clean so maybe try to reference that directly and see if it works.

**Comment by ospfranco** [comment]
Thank you both for commenting! 

@stefanceriu  already tried to manually modify the pod target but headers are still not picked up

**Comment by ospfranco** [comment]
I got it working. I added the entire repo as a sub module since I needed to modify the podspec and created a subspec in my library podspec. But as for getting the headers to work I added `ss.header_mappings_dir = "olm/include"` which preserves the file structure of the headers folder.

![Oscar Franco Screen 000986](https://github.com/matrix-org/olm/assets/1634213/b8f9acc3-3b86-4586-b949-7cfd09ed27f8)

I removed the OLMKit references for the Obj-C headers since I don't need them. Maybe it would be good to add it to the main repo in case anyone else needs to use the sources directly.
